tools/cellxgene_census_builder/src/cellxgene_census_builder/build_soma/experiment_builder.py

```diff
@@ -602,7 +632,22 @@ def _accumulate_X(
 
     if executor is not None:
         return executor.submit(
-            12 * dataset.asset_h5ad_filesize,  # Heuristic value based upon empirical testing.
+            # memory budget:
+            #   X slices: stride * avg_var_nnz * 20 bytes * overhead
+            # + anndata obs/var:  (n_var * 5 cols * 8 + n_obs * 4 * 8) * overhead
+            # + stats space: 5 col * n_obs * 8 bytes * overhead
+            # + working space: fixed value
+            int(
+                (
+                    max(dataset.mean_genes_per_cell, 3000)
+                    * min(ACCUM_X_STRIDE, dataset.dataset_total_cell_count)
+                    * 20
+                    * 8
+                )
+                + (100_000 * 40 + dataset.dataset_total_cell_count * 32) * 8
+                + (dataset.dataset_total_cell_count * 40) * 8
+                + (2 * 1024**3)
+            ),
```

Comment:
just curious, is the "overhead" value a guesstimate? i.e. "assumed inefficiency of data representation due to inefficient typing, copying, etc.". Or is it concretely known to be 8?

Reply:
guesstimate, informed by empirical testing. I've been doing all my test builds on hosts w/o swap, to catch the egregious cases of under-reserving memory.